prog/vm/nexus.rb

```diff
@@ -195,6 +176,39 @@ def allocate
     vm_host_id
   end
 
+  def create_storage_volume_records
+    frame["storage_volumes"].each_with_index do |volume, disk_index|
+      key_encryption_key = if volume["encrypted"]
+        key_wrapping_algorithm = "aes-256-gcm"
+        cipher = OpenSSL::Cipher.new(key_wrapping_algorithm)
+        key_wrapping_key = cipher.random_key
+        key_wrapping_iv = cipher.random_iv
+
+        StorageKeyEncryptionKey.create_with_id(
+          algorithm: key_wrapping_algorithm,
+          key: Base64.encode64(key_wrapping_key),
+          init_vector: Base64.encode64(key_wrapping_iv),
+          auth_data: "#{vm.inhost_name}_#{disk_index}"
+        )
+      end
+
+      VmStorageVolume.create_with_id(
+        vm_id: vm.id,
+        boot: volume["boot"],
+        size_gib: volume["size_gib"],
+        disk_index: disk_index,
+        key_encryption_key_1_id: key_encryption_key&.id
+      )
+    end
+  end
+
+  def remove_storage_volumes_info
```

Comment:
I'd rename this to mention that it's a stack mutation, at a call site it's both kind of alarming and kinda generic and not that terse. `clear_stack_storage_volumes`? Not perfect, but maybe something like that?